### PR TITLE
Added `commands.AutoShardedBot`

### PR DESCRIPTION
**Pull request by CaptBarbossa**
# Description

Made AntiSpamHandler to take `commands.AutoShardedBot` as channel type too

Fixes # (issue)

## Modfication

Please delete options that are not relevant.

- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] Test A
- [x] Test B
Etc...

**Test Configuration**:
* Firmware version:
* Hardware:

# Checklist:

- [x] I have added in logging for all relevant areas 
- [x] I have performed a self-review of my own code
- [x] My code follows the style guidelines of this project
- [x] New and existing unit tests pass locally with my changes
- [x] I have made corresponding changes to the documentation
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works

### CODE REVIEW

**Comment by Skelmis** [comment]
While your at it, care to change channel to bot as well as add discord.Client and discord.AutoShardedClient?

**Comment by CaptBarbossa** [comment]
Not sure what you mean? 🙈

**Comment by Skelmis** [comment]
Add support for all types of bot & client

Also rename the error to say the correct variable

**Comment by CaptBarbossa** [comment]
Sure, will be done and i just figured out tha docs will need an update too...want me to do that? Sorry forgot to check that before, it's my first pull request ever

**Comment by Skelmis** [comment]
Yes you will, there is a checklist in the first message you posted on this pull request of things to tick off that apply to your changes

**Comment by CaptBarbossa** [comment]
I really need to control rushing, I will open the Complete pull request in an hour or so.. Thank you

**Comment by Skelmis** [comment]
No worries. Request a review on this pull request once you feel it's ready

**Comment by CaptBarbossa** [comment]
Should be done, Added the required changes, changed raise message and fixed docs too

File: AntiSpam/AntiSpamHandler.py [hunk 1]
```diff
@@ -137,15 +137,15 @@ class AntiSpamHandler:
     #      and that's how it will be implemented internally most likely
     # TODO Add the ability to toggle dm messages for log messages (To affected users)
 
-    def __init__(self, bot: commands.Bot, **kwargs):
+    def __init__(self, bot: commands.Bot or commands.AutoShardedBot or discord.Client or discord.AutoShardedClient, **kwargs):
```
Comment (Skelmis) [inline AntiSpam/AntiSpamHandler.py:140]: Use typing.Union

**Review by Skelmis — CHANGES_REQUESTED** [review]

**Comment by Skelmis** [comment]
Please also expand the test suite, you can find current examples [here](https://github.com/Skelmis/DPY-Anti-Spam/blob/master/testing/test_ASH.py#L142)

**Comment by CaptBarbossa** [comment]
Done :)

File: testing/test_ASH.py [hunk 2]
```diff
@@ -162,7 +162,7 @@ async def test_botAssignment(self):
             AntiSpamHandler(False)
 
         AntiSpamHandler(commands.Bot(command_prefix="!"))
-        AntiSpamHandler(commands.Bot(command_prefix="!"))
+        AntiSpamHandler(commands.AutoShardedBot(command_prefix="!"))
         AntiSpamHandler(discord.Client)
```
Comment (Skelmis) [inline testing/test_ASH.py:167]: Initialize this class.

File: testing/test_ASH.py [hunk 3]
```diff
@@ -162,7 +162,7 @@ async def test_botAssignment(self):
...
         AntiSpamHandler(discord.AutoShardedClient)
```
Comment (Skelmis) [inline testing/test_ASH.py:168]: Initialize this class.

**Review by Skelmis — CHANGES_REQUESTED** [review]

File: testing/test_ASH.py [hunk 4]
```diff
@@ -162,6 +163,10 @@ async def test_botAssignment(self):
             AntiSpamHandler(False)
 
         AntiSpamHandler(commands.Bot(command_prefix="!"))
+        AntiSpamHandler(commands.AutoShardedBot(command_prefix="!"))
+        discord.Client()
```
Comment (Skelmis) [inline testing/test_ASH.py:167]: You're not testing anything with this line?
Did you mean `AntiSpamHandler(discord.Client())`

File: testing/test_ASH.py [hunk 5]
```diff
@@ -162,6 +163,10 @@ async def test_botAssignment(self):
...
+        discord.AutoShardedClient()
```
Comment (Skelmis) [inline testing/test_ASH.py:168]: You're not testing anything with this line?
Did you mean `AntiSpamHandler(discord.AutoShardedClient())`

**Review by Skelmis — CHANGES_REQUESTED** [review]

File: testing/test_ASH.py (hunk 5 above)
Comment (CaptBarbossa) [reply testing/test_ASH.py:168]: aa yea hmm.. my bad!

**Review by Skelmis — APPROVED** [review]

**Pull request merged**